source/ota_interface.c

```diff
@@ -64,65 +76,87 @@ void setControlInterface( OtaControlInterface_t * pControlInterface )
 OtaErr_t setDataInterface( OtaDataInterface_t * pDataInterface,
                            const uint8_t * pProtocol )
 {
-    OtaErr_t err = OtaErrInvalidDataProtocol;
-    uint32_t i;
-
-    /*
-     * Primary data protocol will be the protocol used for file download if more
-     * than one protocol is selected while creating OTA job.
-     */
-    #if ( configOTA_PRIMARY_DATA_PROTOCOL == OTA_DATA_OVER_MQTT )
-        const char * pProtocolPriority[ OTA_DATA_NUM_PROTOCOLS ] =
+    OtaErr_t err = OtaErrNone;
```

Comment:
maintainability: return status values should always be set to error values, e.g.:

```c
OtaErr_t err = OtaErrInvalidDataProtocol;
```

Reply:
That makes sense. Updated.